syntax = "proto3";

package pbf_db;

import "map_renderer.proto";
import "graph.proto";
import "transport_router.proto";

message Coordinates {
	double lat = 1;
	double lng = 2;
}

message Stop {
	uint64 id = 1;
	string name = 2;
	Coordinates coord = 3;
	map<uint64, int32> road_distances = 4;
}

message Bus {
	string name = 1;
	repeated uint64 route = 2;
	bool is_roundtrip = 3;
}

message TransportCatalogue {
	repeated Stop stop = 1;
	repeated Bus bus = 2;
	RenderSettings render_settings = 3;
	Graph graph = 4;
	Router router = 5;
}
